eclair-core/src/main/scala/fr/acinq/eclair/payment/relay/ChannelRelay.scala

```diff
@@ -118,60 +123,63 @@ class ChannelRelay private(nodeParams: NodeParams,
         context.log.warn(s"couldn't resolve downstream channel $channelId, failing htlc #${upstream.add.id}")
         val cmdFail = CMD_FAIL_HTLC(upstream.add.id, Right(UnknownNextPeer()), commit = true)
         Metrics.recordPaymentRelayFailed(Tags.FailureType(cmdFail), Tags.RelayType.Channel)
-        safeSendAndStop(upstream.add.channelId, cmdFail)
+        safeSendAndStop(upstream.add.channelId, cmdFail, confidence, Some(channelId))
 
       case WrappedAddResponse(addFailed: RES_ADD_FAILED[_]) =>
         context.log.info("attempt failed with reason={}", addFailed.t.getClass.getSimpleName)
-        context.self ! DoRelay
+        context.self ! WrappedConfidence(confidence)
         relay(previousFailures :+ PreviouslyTried(selectedChannelId, addFailed))
 
-      case WrappedAddResponse(_: RES_SUCCESS[_]) =>
+      case WrappedAddResponse(r: RES_SUCCESS[_]) =>
         context.log.debug("sent htlc to the downstream channel")
-        waitForAddSettled()
+        waitForAddSettled(confidence, r.channelId)
     }
 
-  def waitForAddSettled(): Behavior[Command] =
+  def waitForAddSettled(confidence: Double, channelId: ByteVector32): Behavior[Command] =
     Behaviors.receiveMessagePartial {
       case WrappedAddResponse(RES_ADD_SETTLED(_, htlc, fulfill: HtlcResult.Fulfill)) =>
         context.log.debug("relaying fulfill to upstream")
+        Metrics.relaySettleFulfill(confidence)
         val cmd = CMD_FULFILL_HTLC(upstream.add.id, fulfill.paymentPreimage, commit = true)
         context.system.eventStream ! EventStream.Publish(ChannelPaymentRelayed(upstream.amountIn, htlc.amountMsat, htlc.paymentHash, upstream.add.channelId, htlc.channelId, upstream.receivedAt, TimestampMilli.now()))
         recordRelayDuration(isSuccess = true)
-        safeSendAndStop(upstream.add.channelId, cmd)
+        safeSendAndStop(upstream.add.channelId, cmd, confidence, Some(channelId))
 
       case WrappedAddResponse(RES_ADD_SETTLED(_, _, fail: HtlcResult.Fail)) =>
         context.log.debug("relaying fail to upstream")
+        Metrics.relaySettleFail(confidence)
         Metrics.recordPaymentRelayFailed(Tags.FailureType.Remote, Tags.RelayType.Channel)
         val cmd = translateRelayFailure(upstream.add.id, fail)
         recordRelayDuration(isSuccess = false)
-        safeSendAndStop(upstream.add.channelId, cmd)
+        safeSendAndStop(upstream.add.channelId, cmd, confidence, Some(channelId))
     }
 
-  def safeSendAndStop(channelId: ByteVector32, cmd: channel.HtlcSettlementCommand): Behavior[Command] = {
+  def safeSendAndStop(channelId: ByteVector32, cmd: channel.HtlcSettlementCommand, confidence: Double, outgoingChannel_opt: Option[ByteVector32]): Behavior[Command] = {
+    context.log.info("cmd={}, startedAt={}, endedAt={}, confidence={}, originNode={}, outgoingChannel={}", cmd.getClass.getSimpleName, upstream.receivedAt, TimestampMilli.now(), confidence, upstream.receivedFrom, outgoingChannel_opt)
```

Comment:
This is redundant with previous logs in some case, and conflates failure cases that are actually very different. I think we should rather modify the initial log line (`relaying htlc #{} from channelId={} ...`) to include the additional data you're interested in (`confidence` and `originNode`) and modify the other existing logs to include `endedAt`. There is no need to include `outgoingChannel`, it is already logged in `forwarding htlc #{} from channelId={} to channelId={}`.

Reply:
But you're not actually getting the full picture, because there are different failure cases that shouldn't be interpreted similarly at all, and this information is lost when you only have access to `cmd.getClass.getSimpleName`. Whenever we debug a relay case, we grep by `relayId` and look at multiple log lines anyway, that's the only way you can really get a full picture.

The only reason you'd want a single line is if you want to grep all of those lines (for unrelated relays) to do some data analysis (by parsing the log line and extracting the various fields?), is that what you want to do?

Reply:
The idea is exactly to be able to grep all those lines to do some data analysis.

Reply:
Moved it to only log cases when we tried to relay.